grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1381,6 +1387,35 @@ class JsonController {
         render(["aaData": data] as JSON)
     }
 
+    def getSummaryOrderReport = {
+        def location = Location.get(session.warehouse.id)
+
+        String query = """
+            select * 
+            from on_order_summary 
+            where destination_id = :locationId
+            """
+        Sql sql = new Sql(dataSource)
+        def items = sql.rows(query,  [locationId: location.id])
+
+        def data = items.collect {
+            Product product = Product.findByProductCode(it[0].toString())
+            def quantityOnHand = inventoryService.getQuantityOnHand(location, product)
```

Comment:
This is also going to be expensive and unnecessary. You could just get quantity on hand for a list of products. Or for the entire location since that'll be faster than doing each individually.

OR 

again, include this data in the query. 

For example, you could create a view to turn inventory snapshot into a product snapshot and then left outer join that with the on_order_summary table.

Pseudocode:
```
create or replace view product_snapshot as (
select location_id, product_id, sum(quantity)
from inventory_snapshot
group by location_id, product_id)
```
And then you have
```
select *
from on_order_summary 
left outer join product_snapshot on (product_snapshot.product_id = on_order_summary.product_id 
    and product_snapshot.location_id = on_order_summary.location_id)
where destination_id = :locationId
```